File: backend/controllers/userController.js
const User = require('../models/User');

const updateUserStats = async (req, res) => {
    const { userId, subject, correct } = req.body;

    try {
        let user = await User.findOne({ userId });
        if (!user) {
            user = new User({
                userId,
                statistics: {
                    examsTaken: 0,
                    correctAnswers: 0,
                    wrongAnswers: 0,
                    subjectStats: {
                        [subject]: {
                            examsTaken: 0,
                            correctAnswers: 0,
                            wrongAnswers: 0
                        }
                    }
                }
            });
        }

        // Update stats based on answer correctness
        user.statistics.examsTaken++;
        if (correct) {
            user.statistics.correctAnswers++;
            user.statistics.subjectStats[subject].correctAnswers++;
        } else {
            user.statistics.wrongAnswers++;
            user.statistics.subjectStats[subject].wrongAnswers++;
        }

        await user.save();
        res.json(user.statistics);
    } catch (error) {
        res.status(500).json({ error: 'Server error' });
    }
};

module.exports = { updateUserStats };

File: backend/app/controllers/user_controller.py
from flask import jsonify, request
from app.models.user_model import User

def update_user():
    data = request.json
    user_id = data.get("userId")
    stats = data.get("statistics")

    if not user_id or not stats:
        return jsonify({"error": "Invalid data"}), 400

    User.update_user_stats(user_id, stats)
    return jsonify({"message": "User statistics updated"}), 200